pkg/execution/realtime/broadcaster_redis.go

```diff
@@ -75,6 +75,27 @@ func (b *redisBroadcaster) Publish(ctx context.Context, m Message) {
 	}
 }
 
+func (b *redisBroadcaster) PublishStream(ctx context.Context, m Message, data string) {
+	for _, t := range m.Topics() {
+		go func(t Topic) {
+			cmd := b.pubc.B().Publish().Channel(t.String()).Message(
+				string(m.Data) + ":" + data,
+			).Build()
+			for i := 0; i < redisPublishAttempts; i++ {
```

Comment:
Would a log indicating that we ran out of attempts and gave up for this message be relevant if we exhaust all retires?

It'd be nice to distinguish between recoverable retries and full failures.